task/jefe/src/main.rs

```diff
@@ -161,8 +171,33 @@ impl idl::InOrderJefeImpl for ServerImpl<'_> {
                 _msg: &userlib::RecvMessage,
                 index: u8,
             ) -> Result<DumpArea, RequestError<DumpAgentError>> {
-                dump::get_dump_area(self.dump_areas, index)
-                    .map_err(|e| e.into())
+                // If we have cached a dump area, then use it to accelerate
+                // lookup by jumping partway through the linked list
+                let d = if let Some(prev) = self.last_dump_area {
+                    if index == prev.index {
+                        // Easy case: we've already looked up this area
+                        Ok(prev)
+                    } else if let Some(offset) = index.checked_sub(prev.index) {
```

Comment:
i was going to suggest that this expression might be better represented as a `match`, since the two default cases could be combined, but...i think you can't write a match guard like
```rust
Some(prev) if let Some(offset) = index.checked_sub(prev.index) =>
```
which is a bit of a bummer. I suppose you could instead write
```rust
Some(prev) if prev.index >= index => 
```
and then do the subtraction in the body of the match arm, but using `checked_sub` both as the check and the actual subtraction feels much nicer, and the case where the `checked_sub` returns `None` would still have to be handled despite never actually happening, which feels quite sad...